packages/fiber/src/core/hooks.tsx

```diff
@@ -91,43 +92,36 @@ export type LoaderResult<T> = T extends { scene: THREE.Object3D } ? T & ObjectMa
 export type Extensions<T> = (loader: Loader<T>) => void
 
 const memoizedLoaders = new WeakMap<LoaderProto<any>, Loader<any>>()
+const loaderCaches = new Map<Loader<any>, PromiseCache>()
 
 const isConstructor = <T,>(value: unknown): value is LoaderProto<T> =>
   typeof value === 'function' && value?.prototype?.constructor === value
 
-function loadingFn<T>(extensions?: Extensions<T>, onProgress?: (event: ProgressEvent) => void) {
-  return async function (Proto: Loader<T> | LoaderProto<T>, ...input: string[]) {
-    let loader: Loader<any>
-
-    // Construct and cache loader if constructor was passed
-    if (isConstructor(Proto)) {
-      loader = memoizedLoaders.get(Proto)!
-      if (!loader) {
-        loader = new Proto()
-        memoizedLoaders.set(Proto, loader)
-      }
-    } else {
-      loader = Proto
+function prepareLoaderInstance(loader: Loader<any> | LoaderProto<any>): Loader<any> {
+  let loaderInstance: Loader<any>
+
+  // Construct and cache loader if constructor was passed
+  if (isConstructor(loader)) {
+    loaderInstance = memoizedLoaders.get(loader)!
+    if (!loaderInstance) {
+      loaderInstance = new loader()
+      memoizedLoaders.set(loader, loaderInstance)
     }
+  } else {
+    loaderInstance = loader as Loader<any>
+  }
 
-    // Apply loader extensions
-    if (extensions) extensions(loader)
```

Comment:
Is it possible to deopt calls to `useLoader` to only cache for the caller component? Can pass `React.useId` to discern the component.